lib/protocol/ConnectionManagerState.js

```diff
@@ -15,7 +15,7 @@
 
 module.exports = ConnectionManagerState;
 
-var ENCRYPTION_OPTIONS = ['pfx', 'key', 'cert', 'ca', 'passphrase', 'rejectUnauthorized', 'secureProtocol', 'checkServerIdentity'];
+var ENCRYPTION_OPTIONS = ['pfx', 'key', 'cert', 'ca', 'passphrase', 'rejectUnauthorized', 'secureProtocol', 'checkServerIdentity', 'servername'];
```

Comment:
With the whitelist we will always have to keep track of the available properties in net/tls.connect and update the list accordingly. It looks there are more properties missing such as: socket, session, lookup....

Also the name should be SOCKET_OPTIONS or something similar because some of the options there are not related to encryption.

Wouldn't it be easier to pass all options through?